### ISSUE TITLE
help command doesn't complete after the first argument

### ISSUE THREAD

**Issue by minijackson** [epics-base/epics-base#364]
Current situation:

- `help <tab>` completes commands
- `help help <tab>` completes local files.

Not sure how hard it would be to implement.

See https://github.com/epics-base/epics-base/pull/363#pullrequestreview-1334526191

Related to #274

**Comment by AlexanderWells-diamond** [comment]
I've taken a little look at this.

Quoting @mdavidsaver from the linked PR:
> > ... completion doesn't work for the second argument ...
> 
> I did not think of this... May be as simple as changing `arg==1` to `arg>=1` in iocsh.cpp
> 
> ```c
>             } else if(arg==1 && strcmp(def->pFuncDef->name, "help")==0) {
>                 return rl_completion_matches(word, iocsh_complete_command);
> ```

Unfortunately this does not fix it. The issue is a little further up the function, specifically the `err` value that comes from this first if statement:

```
        if(!err) {
            for(arg=1; arg<tokenize.size(); arg++) {
                // BUG: does not work with eg. 'dbpr("X")' as split()
                //      has rewritten lbuf to make this 'dbpr "X"'
                size_t soffset = tokenize.argv[arg]-&lbuf[0];
                size_t eoffset = soffset + strlen(tokenize.argv[arg]);
                if(soffset >= size_t(start) && eoffset <= size_t(end)) {
                    break;
                }
            }
            err = arg-1u >= size_t(def->pFuncDef->nargs);
        }

        if(!err) {
            // we are being asked to complete a valid command,
            // for which we have split argument strings.
```

In the case of running `help dbDumpField dbDump<tab>`, this code exits with an `err` of `true`, which means the tab completion code below does not execute. 

I believe this needs to be modified to account for the variable number of arguments allowed by the `help` function's one and only  argument. It already marks it by having a `iocshArgType type` of `iocshArgArgv`, but this information is unused here. I'm very interested in suggestions for exactly how to accomplish this.

**Comment by mdavidsaver** [comment]
I think I've addressed this with 0c13e6ba6cbae559cb767f63836ca7409000a54d
